### ISSUE TITLE
xpcall is broken

### ISSUE THREAD

**Issue by jonas-schievink** [chucklefish/rlua#36]
rlua REPL session:

```
> xpcall(error, function(...) print("caught error: ", ...) return "errör" end, "testerror")
caught error: 	bad argument #2 to 'error' (number expected, got string)
Boolean(false)	String(String(LuaRef(4)))
> xpcall()
repl: lua/lapi.c:227: lua_copy: Assertion `(((to) != (&luaO_nilobject_))) && "invalid index"' failed.
```

The first one: `error` is not called with the right arguments, it should only get "testerror".
The second one is just a missing check for enough arguments.

`pcall` also has problems:

```
> print(pcall())
false	attempt to call a string value
```

**Comment by kyren** [comment]
Combination of not checking for # of arguments (obviously) and accidental use of lua_insert instead of lua_replace, should be fixed by f9f3d0580458e8af7051095eec16022a5fa1bbba.

Thanks for catching that!

**Comment by jonas-schievink** [comment]
Yep, that fixes it.
